sunpy/net/dataretriever/tests/test_goes_suvi.py

```diff
@@ -0,0 +1,203 @@
+                                         a.Wavelength(131 * u.Angstrom))
+    assert ans2 is True
+    ans3 = suvi_client._can_handle_query(time, a.Instrument('suvi'),
+                                         a.Wavelength(131 * u.Angstrom),
+                                         a.Level(2))
+    assert ans3 is True
+    ans4 = suvi_client._can_handle_query(time)
+    assert ans4 is False
+    ans5 = suvi_client._can_handle_query(time, a.Instrument('aia'))
+    assert ans5 is False
+    ans6 = suvi_client._can_handle_query(time, a.Instrument('suvi'),
+                                         a.goes.SatelliteNumber(16))
+    assert ans6 is True
+
+
+def test_get_goes_sat_num(suvi_client):
+    date = parse_time('2019/06/11 00:00')
+    min_satellite_number = 16  # when SUVI was first included
+    assert suvi_client._get_goes_sat_num(date) >= min_satellite_number
+    assert type(suvi_client._get_goes_sat_num(date)) is int
+
+
+def test_get_goes_sat_num_error(suvi_client):
+    date = parse_time('1800/06/11 00:00')
+    with pytest.raises(ValueError):
+        suvi_client._get_goes_sat_num(date)
+
+
+def test_get_url_for_timerange_errors(suvi_client):
+    """Check that unsupported values raise errors."""
+    tr = TimeRange('2019/06/11 00:00', '2019/06/11 00:10')
+    with pytest.raises(ValueError):
+        suvi_client._get_url_for_timerange(tr, level=0)
+    with pytest.raises(ValueError):
+        suvi_client._get_url_for_timerange(tr, wavelength=100 * u.Angstrom)
+    with pytest.raises(ValueError):
+        suvi_client._get_url_for_timerange(tr, satellitenumber=1)
+
+
+def mock_querry_object(suvi_client, start, end):
+    """
+    Creating a Query Response object and prefilling it with some information
+    """
+    # Creating a Query Response Object
+    obj = {
+        'TimeRange': TimeRange(parse_time(start), parse_time(end)),
+        'Time_start': parse_time(start),
+        'Time_end':  parse_time(end),
+        'source': 'GOES',
+        'instrument': 'SUVI',
+        'physobs': 'flux',
+        'provider': 'NOAA'
+    }
+    results = QueryResponse.create(obj, suvi_client._get_url_for_timerange(TimeRange(start, end)))
+    results.client = suvi_client
+    return results
+
+
+@pytest.mark.remote_data
+def test_fetch_working(suvi_client):
+    """
+    Tests if the online server for fermi_gbm is working.
+    This also checks if the mock is working well.
+    """
+    start = '2019/05/25 00:50'
+    end = '2019/05/25 00:52'
+    qr1 = suvi_client.search(a.Time(start, end), a.Instrument('suvi'))
+
+    # Mock QueryResponse object
+    mock_qr = mock_querry_object(suvi_client, start, end)
+
+    # Compare if two objects have the same attribute
+
+    mock_qr = mock_qr[0]
+    qr = qr1[0]
+
+    assert mock_qr.source == qr.source
+    assert mock_qr.provider == qr.provider
+    assert mock_qr.physobs == qr.physobs
+    assert mock_qr.instrument == qr.instrument
+    assert mock_qr.url == qr.url
+    assert mock_qr.time == qr.time
```

Comment:
this time check is now failing - I think its now because the time timedates from `get_time_for_url`?